homeassistant/loader.py

```diff
@@ -71,7 +71,9 @@ def get_component(hass,  # type: HomeAssistant
     Async friendly.
     """
     try:
-        return hass.data[DATA_KEY][comp_or_platform]  # type: ignore
+        comp = hass.data[DATA_KEY][comp_or_platform]  # type: ignore
+        if comp is not None:
```

Comment:
How can this be None?

Reply:
https://github.com/home-assistant/home-assistant/blob/48af5116b34f1d19c5be2f913ae06831e871034f/homeassistant/setup.py#L165

Only place used `loader.set_component` method. Is that a legacy code?

Reply:
That happens if a component is not correctly coded, we don't have such components in our code base. It's only when people are having badly written custom components, hence the warning.